core/dbt/parser/models.py

```diff
@@ -196,42 +195,47 @@ def get_compiled_path(cls, block: FileBlock):
         return block.path.relative_path
 
     def parse_python_model(self, node, config, context):
+        config_keys_used = []
+        config_keys_defaults = []
+
         try:
             tree = ast.parse(node.raw_code, filename=node.original_file_path)
         except SyntaxError as exc:
             msg = validator_error_message(exc)
             raise ParsingException(f"{msg}\n{exc.text}", node=node) from exc
 
-        # We are doing a validator and a parser because visit_FunctionDef in parser
-        # would actually make the parser not doing the visit_Calls any more
-        dbtValidator = PythonValidationVisitor()
-        dbtValidator.visit(tree)
-        dbtValidator.check_error(node)
+        # Only parse if AST tree has instructions in body
+        if tree.body:
```

Comment:
One question related to this, you can still reference this model, and we will try to materialize it if you do dbt run. Should we somehow skip the current model?